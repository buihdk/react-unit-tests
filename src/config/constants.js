export const FLIGHT_SORT_BY_QUERY = {
  none: 'FlightSortByNone',
  name: 'FlightSortByName',
  startDate: 'FlightSortByStartDate',
  endDate: 'FlightSortByEndDate',
};

export const ORDER_BY_QUERY = {
  none: 'OrderByNone',
  asc: 'OrderByAsc',
  desc: 'OrderByDesc',
};

export const MAX_LIFETIME_LIMIT = 2147483647;

export const STANDARD_IMAGE = {
  maxWidth: 1200,
  maxHeight: 1200,
  type: 'image/jpeg',
  quality: 0.7,
  safeSizeInBytes: 250000,
};

export const URL_REGEX = /.*/;

export const MAX_LINK_URL = 2048;

export const FOOD_BANNER_AD = 'foodBannerAd';

export const SIMPLE_NATIVE_CONTENT_AD = 'simpleNativeContentAd';

export const FOOD_BANNER_IMAGE_RATIO_7X3 = 7 / 3;

export const DEFAULT_FOOD_BANNER_IMAGE_RATIO = 2.5;
