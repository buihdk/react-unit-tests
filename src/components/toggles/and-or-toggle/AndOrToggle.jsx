import React, { memo } from 'react';
import I18n from 'i18n-js';
import PropTypes from 'prop-types';
import { Radio, message as Message } from 'antd';

import { errorMessageDuration } from 'src/config/constants';
import CustomAlert from 'src/components/alert/CustomAlert';

const styles = {};

const AndOrToggle = ({
  form: { getFieldDecorator },
  type,
  initialLogicalOperator,
}) => (
  <div className={styles.wrapper}>
    <CustomAlert
      message={I18n.t(`messages.flights.${type}Targeting.message`)}
      description={I18n.t(`messages.flights.${type}Targeting.description`)}
    />
    <div className={styles.toggleWrapper}>
      <span className={styles.label}>{I18n.t('common.select')}</span>
      {getFieldDecorator(`${type}LogicalOperatorToggle`, {
        initialValue: initialLogicalOperator,
      })(
        <Radio.Group
          buttonStyle="solid"
          onChange={event =>
            Message.warn(
              I18n.t(`messages.${type}ToggleChange`, {
                value: event.target.value.toUpperCase(),
              }),
              errorMessageDuration,
            )
          }
        >
          <Radio.Button buttonStyle="secondary" value="and">
            {I18n.t('common.and').toUpperCase()}
          </Radio.Button>
          <Radio.Button buttonStyle="secondary" value="or">
            {I18n.t('common.or').toUpperCase()}
          </Radio.Button>
        </Radio.Group>,
      )}
      <div>
        <span className={styles.note}> {I18n.t('messages.note')} </span>
        {I18n.t(`messages.${type}Note`)}
      </div>
    </div>
  </div>
);

AndOrToggle.propTypes = {
  form: PropTypes.shape({
    getFieldDecorator: PropTypes.func.isRequired,
  }).isRequired,
  type: PropTypes.string.isRequired,
  initialLogicalOperator: PropTypes.string,
};

AndOrToggle.defaultProps = {
  initialLogicalOperator: '',
};

export default memo(AndOrToggle);
