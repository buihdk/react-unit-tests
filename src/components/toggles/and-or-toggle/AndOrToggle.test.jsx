import React from 'react';
import { message as Message } from 'antd';
import { shallow, mount } from 'enzyme';
import I18n from 'i18n-js';
import toJSON from 'enzyme-to-json';

import { errorMessageDuration } from 'src/config/constants';

import AndOrToggle from './AndOrToggle';

const MessageWarn = jest.spyOn(Message, 'warn');

const props = {
  form: {
    getFieldDecorator: () => element => element,
  },
  type: 'properties',
};

describe('AndOrToggle', () => {
  test('renders without crashing ', () => {
    const wrapper = shallow(<AndOrToggle {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test(`simulate to change to 'OR'`, () => {
    const wrapper = mount(<AndOrToggle {...props} />);
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { target: { value: 'or' } });

    expect(MessageWarn).toBeCalledWith(
      I18n.t('messages.propertiesToggleChange', { value: 'OR' }),
      errorMessageDuration,
    );
  });
});
