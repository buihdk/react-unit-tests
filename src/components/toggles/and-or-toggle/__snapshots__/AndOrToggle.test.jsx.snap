// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AndOrToggle renders without crashing  1`] = `
<div>
  <CustomAlert
    className=""
    description="[missing \\"en.messages.flights.propertiesTargeting.description\\" translation]"
    message="[missing \\"en.messages.flights.propertiesTargeting.message\\" translation]"
  />
  <div>
    <span>
      Select
    </span>
    <Memo([object Object])
      buttonStyle="solid"
      onChange={[Function]}
    >
      <ForwardRef(RadioButton)
        buttonStyle="secondary"
        value="and"
      >
        AND
      </ForwardRef(RadioButton)>
      <ForwardRef(RadioButton)
        buttonStyle="secondary"
        value="or"
      >
        OR
      </ForwardRef(RadioButton)>
    </Memo([object Object])>
    <div>
      <span>
         
        [missing "en.messages.note" translation]
         
      </span>
      [missing "en.messages.propertiesNote" translation]
    </div>
  </div>
</div>
`;
